newIDE/app/src/InstancesEditor/index.js

```diff
@@ -636,23 +636,29 @@ export default class InstancesEditor extends Component<Props> {
     this.lastCursorY = y;
   };
 
-  _onBackgroundClicked = (x: number, y: number) => {
+  _onBackgroundClicked = (x: number, y: number, event?: PointerEvent) => {
     this.lastCursorX = x;
     this.lastCursorY = y;
     this.pixiRenderer.view.focus();
 
+    // KeyboardShortcuts.shouldMoveView cannot be used here because
+    // the click event fires first on the background, then on the pixi
+    // view which KeyboardShortcuts listens to. So KeyboardShortcuts
+    // will always be late.
+    const shouldMoveView = event ? event.button === MID_MOUSE_BUTTON : false;
```

Comment:
This is why I think a better design of the KeyboardShortcuts should have taken the `event` as an argument